packages/babel-plugin/src/utils/ast-builders.tsx

```diff
@@ -325,6 +377,19 @@ const styledTemplate = (opts: StyledTemplateOpts, meta: Metadata): t.Node => {
       })
     : t.identifier('style');
 
+  const unconditionalClassNames: string[] = [];
+  let logicalClassNames = '';
+
+  opts.classNames.forEach((item) => {
+    if (t.isStringLiteral(item)) {
+      unconditionalClassNames.push(item.value);
+    } else if (t.isLogicalExpression(item)) {
+      logicalClassNames += buildStyledConditionalClasses(item);
```

Comment:
I believe you could just remove the entire `buildStyledConditionalClasses` in favour of using the raw source code, since it has already been replaced with the class.

So it would be something like:

```
import generate from "@babel/generator";

logicalClassNames += generate(item).code;
```

Otherwise, you could build the source into `item`, but you may need to add `source` mutations throughout the code to make sure it's in sync with any replacements.

Reply:
Hey @MonicaOlejniczak! I can't fully remember TBH but iirc it was one to reduce the total amount of generated CSS (see: https://github.com/atlassian-labs/compiled/pull/379/files#diff-e7dbf716e5c9790d9d71cc4825026c86ad8c49d6edfd30957ebe37d069b55540R44)